.github/actions/secure-upload-folder/action.yml

```
@@ -0,0 +1,32 @@
+name: "Secure folder upload"
+description: "Upload a folder as a tarball and outputs its SHA256"
+inputs:
+  name:
+    description: "Artifact name."
+    default: "artifact"
+    required: false
+  path:
+    description: 'Path to the folder'
+    required: true
+outputs:
+  sha256:
+    description: "SHA256 of the uploaded tarball."
+    value: "${{ steps.upload.outputs.sha256 }}"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Create tarball
+      env:
+        PATH: ${{ inputs.path }}
+      run: |
+        tar -cvf "$PATH" "$PATH.tar"
```

Do you not want to compress it? Also, the arguments are reversed.

```suggestion
        tar -czvf "$PATH.tgz" "$PATH"
```